core/gsb-api/src/service.rs

```diff
@@ -0,0 +1,447 @@
+            .iter()
+            .rev()
+            .collect()
+    }
+}
+
+impl From<Bind> for Service {
+    fn from(bind: Bind) -> Self {
+        let msg_handler = MessagesHandling::Buffering(Rc::default());
+        // convert to error and return it when e.g. components empty
+        let addr_prefix = bind.addr_prefix;
+        let mut addresses = HashSet::new();
+        for component in bind.components {
+            addresses.insert(format!("{addr_prefix}/{component}"));
+        }
+        Service {
+            addr_prefix,
+            addresses,
+            msg_handler,
+        }
+    }
+}
+
+impl Actor for Service {
+    type Context = Context<Self>;
+
+    fn started(&mut self, ctx: &mut Self::Context) {
+        _ = ya_service_bus::actix_rpc::bind_raw(&self.addr_prefix, ctx.address().recipient());
+    }
+}
+
+#[derive(Message, Debug)]
+#[rtype(result = "()")]
+pub(crate) struct DropMessages {
+    pub(crate) reason: CloseReason,
+}
+
+impl Handler<DropMessages> for Service {
+    type Result = ResponseFuture<<DropMessages as Message>::Result>;
+
+    fn handle(&mut self, msg: DropMessages, _: &mut Self::Context) -> Self::Result {
+        let disconnect_future = self.msg_handler.disconnect(msg);
+        Box::pin(async { disconnect_future.await })
+    }
+}
+
+impl Handler<RpcRawCall> for Service {
+    type Result = ResponseFuture<Result<Vec<u8>, GsbError>>;
+
+    fn handle(&mut self, msg: RpcRawCall, _ctx: &mut Self::Context) -> Self::Result {
+        let addr = msg.addr;
+        log::debug!("Incoming GSB RAW call (addr: {addr})");
+
+        if !self.addresses.contains(&addr) {
+            //TODO use futures::ready! or sth
+            return Box::pin(async move {
+                Err(GsbError::GsbBadRequest(format!(
+                    "No supported msg type for addr: {addr}"
+                )))
+            });
+        }
+        let id = uuid::Uuid::new_v4().to_string();
+        log::debug!("Msg addr: {addr}, id: {id}");
+        let component = Service::addr_prefix_to_component(&addr);
+        let msg = WsRequest {
+            component,
+            id,
+            payload: msg.body,
+        };
+        let msg_handling_future = self.msg_handler.handle_request(msg);
+        Box::pin(async {
+            let receiver = match msg_handling_future.await {
+                Ok(receiver) => receiver,
+                Err(err) => {
+                    log::error!("Sending error (runtime) GSB response: {err}");
+                    return Err(GsbError::GsbFailure(err.to_string()));
+                }
+            };
+            let ws_response = match receiver.await {
+                Ok(ws_response) => ws_response,
+                Err(err) => {
+                    log::error!("Sending error (internal) GSB response: {err}");
+                    return Err(GsbError::GsbFailure(err.to_string()));
+                }
+            };
+            log::info!("Sending GSB response: {ws_response:?}");
```

Comment:
This message has no broader context. There are hundred of place in yagna, where you send GSB response, so you don't see where this happened, for what reason and in what context.
So for reader this log is meaningless.

The second problem is too high log level.

Reply:
Changed to debug and updated it to print only a message `id` because it gets printed previously (instead of printing whole response).
Logger prints module name, so there will be some context.
All these messages are mainly a side effect of my debugging of the feature.

Reply:
Logs are not well thought over but I changed all which could print e.g. debug value of payload bytes, so it instead prints id or address prefix.